core/libdivecomputer.c

```diff
@@ -167,13 +162,26 @@ static int parse_gasmixes(device_data_t *devdata, struct dive *dive, dc_parser_t
 				cyl.type.workingpressure.mbar = lrint(tank.workpressure * 1000);
 
 				cyl.cylinder_use = OC_GAS;
+#if DC_VERSION_CHECK(0,9,0)
```

Comment:
These version checks add a lot of complexity to the code. Can't we just build against a fixed version (which is what we are doing anyway), and avoid the added complexity?

Reply:
The api changes in libdivecomputer v0.9.0 are not backwards compatible. So without the version checks you can either support the v0.8.0 or the v0.9.0 api, but not both. The subsurface fork is currently at v0.8.0, so if someone already wants to try some of the v0.9.0 changes now, the build will fail without these patches. Of course, once the v0.9.0 api gets merged in the subsurface fork, the version checks won't be needed anymore, because at that point the v0.8.0 compatibility can be dropped.

Anyway, even if this PR is not accepted as-is, the code can still serve as an example for what is needed to move to v0.9.0 api. I already had the code, so why not share it?

Reply:
I think they're small enough for now, and it makes it easy to yank them out in the future when the 0.8.0 api is irrelevant.

Reply:
No. We will not add these version checks back.
Subsurface very intentionally builds against the bundled version of libdc in order to avoid this mess.

Reply:
> The subsurface fork is currently at v0.8.0, so if someone already wants to try some of the v0.9.0 changes now, the build will fail without these patches.

I think the changes in Subsurface that are needed to actually make use of the API / data format changes in libdivecomputer 0.9 will have to go deeper than this - right down to a change of the data model, to treat and store gasmixes and tank sensor information as separate groups of entities, and the addition of a UI to allow the user to define and maintain mappings between the two that match their gear configuration(s).

I can probably find some time to start on this, and once this is ready this can be merged together with an update of libdivecomputer to 0.9. But in my opinion this should be after the next release of Subsurface has been completed.

As a side note, in [SemVer](https://semver.org/#summary), breaking API changes are typically denoted in a _major_ version number change - following this pattern will make it easier for users of the library to understand what is going on.